ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/presenter/ToolAdminPresenterImpl.java

```diff
@@ -0,0 +1,221 @@
+                            result.show();
+                        }
+                    });
+                } else {
+                    ErrorHandler.post(caught);
+                }
+            }
+
+            @Override
+            public void onSuccess(Void result) {
+                announcer.schedule(new SuccessAnnouncementConfig(appearance.deleteToolSuccessText()));
+                listStore.remove(listStore.findModelWithKey(event.getTool().getId()));
+            }
+        });
+    }
+
+    @Override
+    public void onToolSelected(ToolSelectedEvent event) {
+        toolAdminServiceFacade.getToolDetails(event.getTool().getId(), new AsyncCallback<Tool>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(Tool result) {
+                view.editToolDetails(result);
+            }
+        });
+    }
+
+    @Override
+    public void onSaveToolSelected(SaveToolSelectedEvent event) {
+        updateTool(event.getTool(), false);
+    }
+
+    void updateTool(final Tool tool, final boolean overwrite) {
+        toolAdminServiceFacade.updateTool(tool, overwrite, new AsyncCallback<Void>() {
+            @Override
+            public void onFailure(final Throwable caught) {
+                String serviceError = getServiceError(caught);
+                if (serviceError.equals(ServiceErrorCode.ERR_NOT_WRITEABLE.toString())) {
+                    overwriteAppDialog.get(new AsyncCallback<OverwriteAppDialog>() {
+                        @Override
+                        public void onFailure(Throwable caught) {
+                            ErrorHandler.post(caught);
+                        }
+
+                        @Override
+                        public void onSuccess(OverwriteAppDialog result) {
+                            result.setText(caught);
+                            result.show();
+                            result.addOkButtonSelectHandler(new SelectEvent.SelectHandler() {
+                                @Override
+                                public void onSelect(SelectEvent event) {
+                                    updateTool(tool, true);
+                                }
+                            });
+                        }
+                    });
+
+                } else {
+                    ErrorHandler.post(caught);
+                }
+            }
+
+            @Override
+            public void onSuccess(Void result) {
+                announcer.schedule(new SuccessAnnouncementConfig(appearance.updateToolSuccessText()));
+                updateView();
+            }
+        });
+    }
+
+    String getServiceError(Throwable caught) {
+        SimpleServiceError simpleServiceError =  AutoBeanCodex.decode(factory, SimpleServiceError.class, caught.getMessage())
+                     .as();
+        return simpleServiceError.getErrorCode();
+    }
```

Comment:
Not all errors will return a JSON body, so attempting to parse JSON here may cause an exception in the UI.

For example, I didn't update my `de.properties` with the tool admin endpoint, so when attempting to save a tool, the server-side returned a `500 Internal Server Error The call failed on the server; see server log for details` string, which caused this method to throw a SyntaxError exception, but in the UI the form closed and no error was displayed to the user.